src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/SolutionExplorerInProcess.cs

```diff
@@ -553,21 +554,47 @@ private async Task<string> WaitForBuildToFinishAsync(IVsOutputWindowPane buildOu
         {
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            await KnownUIContexts.SolutionExistsAndNotBuildingAndNotDebuggingContext;
+            var buildManager = await GetRequiredGlobalServiceAsync<SVsSolutionBuildManager, IVsSolutionBuildManager2>(cancellationToken);
+            using var semaphore = new SemaphoreSlim(1);
+            using var solutionEvents = new UpdateSolutionEvents(buildManager);
+            await semaphore.WaitAsync();
+            void HandleUpdateSolutionDone(bool succeeded, bool modified, bool canceled) => semaphore.Release();
```

Comment:
we don't seem to use any of this data.  i'm not sure why we need all this callback info just to release a semaphore.  we can just produce a TCS that is triggered once the buildmanager reports the build is done.